src/expr/src/expr/expr_regexp.rs

```diff
@@ -248,3 +251,377 @@ impl Expression for RegexpMatchExpression {
+                                        &DataType::from(flag_node.get_return_type().unwrap()),
+                                    )
+                                    .map_err(|e| ExprError::Internal(e.into()))?;
+
+                                    match flag_datum {
+                                        Some(ScalarImpl::Utf8(v)) => v.to_string(),
+                                        // NULL replacement
+                                        // FIXME: Do we need the NULL match arm here?
+                                        _ => bail!("Expected flag to be a String"),
+                                    }
+                                }
+                                _ => {
+                                    return Err(ExprError::UnsupportedFunction(
+                                        "non-constant in regexp_replace".to_string(),
+                                    ))
+                                }
+                            };
+                        }
+                    } else {
+                        flags = placeholder;
+                    }
+                }
+            }
+        }
+
+        // TODO: Any other error handling?
+        if let Some(_other) = children.next() {
+            // There should not any other option after the `flags`
+            bail!("invalid parameters specified in regexp_replace");
+        }
+
+        // Construct the final `RegexpReplaceExpression`
+        let ctx = RegexpContext::new(&pattern, &flags)?;
+
+        let mut global_flag = false;
+        if !flags.is_empty() && flags.starts_with('g') {
+            // Set the `global_flag` to true
+            global_flag = true;
+        }
+
+        Ok(Self {
+            source,
+            ctx,
+            replacement,
+            return_type,
+            start,
+            n,
+            global_flag,
+        })
+    }
+}
+
+impl RegexpReplaceExpression {
+    /// Match and replace one row, return the replaced string
+    fn match_row(&self, text: Option<&str>) -> Option<String> {
+        if let Some(text) = text {
+            // The start position to begin the search
+            let start = if self.start != -1 { self.start - 1 } else { 0 };
+
+            if self.global_flag {
+                // `-g` enabled, we need to replace all the occurrence of the matched pattern
+                if self.ctx.0.captures_len() <= 1 {
+                    println!("Path One");
+                    // There is no capture groups in the regex
+                    return Some(
+                        self.ctx
+                            .0
+                            .replace_all(&text[start as usize..], self.replacement.clone())
+                            .into(),
+                    );
+                } else {
+                    println!("Path Two");
+
+                    // Get the replaced string
+                    let regex = Regex::new(r"\\([1-9])").unwrap();
```

Comment:
I haven’t understood the purpose of this regular expression yet, but at least the expensive repetitive Regex construction should not be done in match_row. Please move this Regex to RegexpReplaceExpression.

Reply:
This is used to match the expression like `X\nY` and replace this to `X${n}Y` in order to make use of the `expand` substitution.
But yes, I'll move this to `RegexpReplaceExpression`